packages/demos/stories/TableColumnsDragDrop.tsx

```diff
@@ -0,0 +1,215 @@
+/**
+ * SPDX-FileCopyrightText: © 2020 Liferay, Inc. <https://liferay.com>
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import ClayTable from '@clayui/table';
+import {XYCoord} from 'dnd-core';
+import React from 'react';
+import {DropTargetMonitor, useDrag, useDrop} from 'react-dnd';
+
+interface IDraggableTableHeadingProps {
+	expanded: boolean;
+	id: string;
+	/**
+	 * Index of the List Item in order to keep track of it's positioning within the array
+	 */
+	index: number;
+
+	/**
+	 * Function that handles the moving of List Items
+	 */
+	onMove: (dragIndex: number, hoverIndex: number) => void;
+	title: string;
+}
+
+interface IDragItem {
+	/**
+	 * List Item unique identifier
+	 */
+	id: string;
+
+	/**
+	 * Index of the List Item in order to keep track of it's positioning within the array
+	 */
+	index: number;
+
+	/**
+	 * Type of data that is allowed to be dragged
+	 */
+	type: string;
+}
+
+const DraggableTableHeaderCell: React.FunctionComponent<
+	IDraggableTableHeadingProps
+> = ({expanded, id, index, onMove, title}) => {
+	const ref = React.useRef<HTMLTableHeaderCellElement>(null);
+
+	const [, drop] = useDrop({
+		accept: 'tableColumn',
```

Comment:
This is something that we define, it's purpose is to prevent elements being dropped in places they aren't supposed to be dropped. For example, dropping a table row where a heading is supposed to be dropped.